test/e2e/sync_lb_fail_test.go

```diff
@@ -0,0 +1,125 @@
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var icName = types.NamespacedName{Namespace: "openshift-ingress", Name: "synclb"}
+
+var badLBService = corev1.Service{
+	ObjectMeta: metav1.ObjectMeta{
+		Name:      "router-" + icName.Name,
+		Namespace: "openshift-ingress",
+		Annotations: map[string]string{
+			// this annotation is intentionally set to an invalid value, which
+			// should cause a SyncLoadBalancerFailed event
+			"service.beta.kubernetes.io/aws-load-balancer-proxy-protocol": "nonsense",
+		},
+		Labels: map[string]string{
+			"app": "router",
+			"ingresscontroller.operator.openshift.io/owning-ingresscontroller": icName.Name,
+			"router": "router-" + icName.Name,
+		},
+	},
+	Spec: corev1.ServiceSpec{
+		Type: corev1.ServiceTypeLoadBalancer,
+		Selector: map[string]string{
+			"ingresscontroller.operator.openshift.io/deployment-ingresscontroller": icName.Name,
+		},
+		Ports: []corev1.ServicePort{
+			{
+				Name:       "http",
+				Port:       80,
+				Protocol:   corev1.ProtocolTCP,
+				TargetPort: intstr.IntOrString{IntVal: 80},
+			},
+			{
+				Name:       "https",
+				Port:       443,
+				Protocol:   corev1.ProtocolTCP,
+				TargetPort: intstr.IntOrString{IntVal: 443},
+			},
+		},
+	},
+}
+
+func TestSyncLBFailEvent(t *testing.T) {
+	if infraConfig.Status.Platform != configv1.AWSPlatformType {
+		t.Skip("test skipped on non-aws platforms")
+		return
+	}
+	if err := kclient.Create(context.TODO(), &badLBService); err != nil {
+		t.Fatalf("failed to create service: %v", err)
+	}
+	defer kclient.Delete(context.TODO(), &badLBService)
+	ic := newLoadBalancerController(icName, icName.Name+dnsConfig.Spec.BaseDomain)
+	if err := kclient.Create(context.TODO(), ic); err != nil {
+		t.Fatalf("failed to create ingresscontroller: %v", err)
+	}
+	defer kclient.Delete(context.TODO(), ic)
+	eventList := &corev1.EventList{}
+	err := wait.PollImmediate(1*time.Second, 5*time.Minute, func() (bool, error) {
+		if err := kclient.List(context.TODO(), eventList, client.InNamespace(badLBService.ObjectMeta.Namespace)); err != nil {
+			return false, nil
+		}
+		for _, event := range eventList.Items {
+			if event.Reason == "SyncLoadBalancerFailed" {
+				return true, nil
+			}
+		}
+		return false, nil
+	})
+	if err != nil {
+		t.Fatalf("Timed out waiting for SyncLoadBalancerFailed event")
+	}
+	expectedConditions := []operatorv1.OperatorCondition{
+		{Type: operatorv1.LoadBalancerReadyIngressConditionType, Status: operatorv1.ConditionFalse},
+		{Type: operatorv1.OperatorStatusTypeDegraded, Status: operatorv1.ConditionTrue},
+	}
+	if err := waitForIngressControllerCondition(t, kclient, 5*time.Minute, icName, expectedConditions...); err != nil {
+		t.Fatalf("Timed out waiting for status conditions LoadBalancerReady=false and Degraded=true for ingresscontroller %s", icName.Name)
+	}
+	err = wait.PollImmediate(1*time.Second, 5*time.Minute, func() (bool, error) {
```

Comment:
another block comment here would be super helpful, this block of code is taking me a while to interpret.